### ISSUE TITLE
State transition are only created when running terraform apply twice. 

### ISSUE THREAD

**Issue by phofmann** [labd/terraform-provider-commercetools#329]
**Information**
Terraform version: 1.3.5
Provider version: 1.5.1

**To Reproduce**
Steps to reproduce the behavior: Create states and transitions. Run terraform apply to create them. The transitions are only created when running terraform apply for the second time. 
Not run terraform destroy. You will receive errors like:
│ Error: Object 1bfd0a5d-ca8e-4b6f-8d21-21065d372855 has a different version than expected. Expected: 3 - Actual: 4.
│ Error: Object d3e4100b-e95b-41dc-b6b7-f8533803dfa3 has a different version than expected. Expected: 2 - Actual: 3.
│ Error: Object b130ef6e-854e-4551-bdf5-2d378b55c172 has a different version than expected. Expected: 3 - Actual: 4.
│ Error: Object d800fc69-996d-40d3-a6d0-07baf086815e has a different version than expected. Expected: 3 - Actual: 4.

I tried adding depends_on but did not succeed either.

```hcl
resource "commercetools_state" "product_in_creation" {
  key  = "product-in-creation"
  type = "ProductState"
  name = {
    en = "In creation"
    de = "Wird erzeugt"
  }
  description = {
    en = "In creation"
    de = "Wird erzeugt"
  }
  initial = true
}

resource "commercetools_state" "product_ready_to_be_translated" {
  key  = "product-ready-to-be-translated"
  type = "ProductState"
  name = {
    en = "Ready to be translated"
    de = "Bereit übersetzt zu werden"
  }
  description = {
    en = "Ready to be translated"
    de = "Bereit übersetzt zu werden"
  }
  initial = false
}

resource "commercetools_state" "product_in_translation" {
  key  = "product-in-translation"
  type = "ProductState"
  name = {
    en = "In translation"
    de = "Wird übersetzt"
  }
  description = {
    en = "In translation"
    de = "Wird übersetzt"
  }
  initial = false
}

resource "commercetools_state" "product_ready_to_be_reviewed" {
  key  = "product-ready-to-be-reviewed"
  type = "ProductState"
  name = {
    en = "Ready to be reviewed"
    de = "Bereit zur Überprüfung"
  }
  description = {
    en = "Ready to be reviewed"
    de = "Bereit zur Überprüfung"
  }
  initial = false
}

resource "commercetools_state" "product_in_review" {
  key  = "product-in-review"
  type = "ProductState"
  name = {
    en = "In review"
    de = "Wird überprüft"
  }
  description = {
    en = "In review"
    de = "Wird überprüft"
  }
  initial = false
}

resource "commercetools_state" "product_ready_for_publication" {
  key  = "product-ready-for-publication"
  type = "ProductState"
  name = {
    en = "Ready to be published"
    de = "Bereit zur Publikation"
  }
  description = {
    en = "Ready to be published"
    de = "Bereit zur Publikation"
  }
  initial = false
}

// Only allow transition from ready to be translated to ready to be reviewed

resource "commercetools_state_transitions" "transition_0" {
  from = commercetools_state.product_in_creation.id
  to = [
    commercetools_state.product_ready_to_be_translated.id,
  ]
  depends_on = [
    commercetools_state.product_in_creation,
    commercetools_state.product_ready_to_be_translated
  ]
}

resource "commercetools_state_transitions" "transition_1" {
  from = commercetools_state.product_ready_to_be_translated.id
  to = [
    commercetools_state.product_in_translation.id,
  ]
  depends_on = [
    commercetools_state.product_ready_to_be_translated,
    commercetools_state.product_in_translation
  ]
}

resource "commercetools_state_transitions" "transition_2" {
  from = commercetools_state.product_in_translation.id
  to = [
    commercetools_state.product_ready_to_be_reviewed.id,
  ]
  depends_on = [
    commercetools_state.product_in_translation,
    commercetools_state.product_ready_to_be_reviewed
  ]
}

resource "commercetools_state_transitions" "transition_3" {
  from = commercetools_state.product_ready_to_be_reviewed.id
  to = [
    commercetools_state.product_in_review.id,
  ]
  depends_on = [
    commercetools_state.product_ready_to_be_reviewed,
    commercetools_state.product_in_review
  ]
}

resource "commercetools_state_transitions" "transition_5" {
  from = commercetools_state.product_in_review.id
  to = [
    commercetools_state.product_ready_to_be_translated.id,
    commercetools_state.product_ready_for_publication.id,
  ]
  depends_on = [
    commercetools_state.product_ready_to_be_translated,
    commercetools_state.product_ready_for_publication
  ]
}


// Disable transitions from product_ready_to_be_reviewed to other
resource "commercetools_state_transitions" "transition_final" {
  from = commercetools_state.product_ready_for_publication.id
  to   = []
  depends_on = [
    commercetools_state.product_ready_for_publication,
  ]
}

```

**Comment by mvantellingen** [comment]
Can confirm that I get some issues with the given example. Will look into it for next version

**Comment by mvantellingen** [comment]
Should be fixed in 1.6.2. Let me know otherwise and we can investigate :-)

**Comment by phofmann** [comment]
Works perfectly fine now. Thank you.
